libr/bin/p/bin_elf.inc

```diff
@@ -1098,6 +1098,11 @@ static RBinInfo* info(RBinFile *bf) {
 		return NULL;
 	}
 	ret->arch = str;
+	
+	if (!strcmp (ret->arch, "mips")) {   
```

Comment:
Instead of doing this only for `mips`, I would do it no matter what. It seems like `Elf_(r_bin_elf_get_head_flag)` is being abused to get the type of CPU, while, in theory, it could hold any type of arch-dependent flags. I would create a new function (e.g. `Elf_(r_bin_elf_get_cpu)`) and use it here. At the moment, it could just return NULL/"" for all architectures except MIPS, where it would use the head_flags to determine it. In the future that could be extended to determine the cpu type in other cases as well, but I'm not sure it is good to assume that head_flags == cpu in all cases.

Reply:
The flag does exist in the ELF head
I just think that some archs  may need it under certain circumstances, or need it in the future.

Reply:
It is ok to get that info from the ELF header flags field, however that field may contain even additional information as well. So what I'm saying is that we should have a separate function that returns the `cpu` identified from the file format (how that function detects the cpu may be different based on the architectures, etc. In case of MIPS, for example, it can just parse the flags header as it is doing now).

Reply:
i got you